tests/unit/test_state.py

```diff
@@ -71,6 +71,24 @@ def test_from_charm_invalid_charm_config(harness: ops.testing.Harness):
         state.State.from_charm(charm=harness.charm)
 
 
+def test_from_charm_invalid_server_url(
+    harness: ops.testing.Harness, config: typing.Dict[str, str]
+):
+    """
+    arrange: given charm configuration data with invalid server_url attribute
+    act: when the state is initialized from_charm.
+    assert: InvalidStateError is raised. (due to pydantic's AnyHttpUrl validator)
+    """
+    invalid_config = config
+    # This configuration is invalid because schema (http or https) must be specified
+    invalid_config["jenkins_url"] = "example.com"
+    harness.update_config(invalid_config)
```

Comment:
You won't be leveraging the harness so, can you mock the charm instead, as in
```
mock_charm = unittest.mock.MagicMock(spec=ops.CharmBase)
mock_charm.config = {"rjenkins_url": "example.com"}
```

Reply:
Could you also fix the typo in the method name in line 92?

Reply:
I fixed the typo

Reply:
Let's keep it as it is for now, then. But I'm not very keen on having the state relying on charm features too